pkg/kube/inject/testdata/inject/auth.non-default-service-account.yaml.injected

```diff
@@ -17,16 +17,14 @@ spec:
         prometheus.io/path: /stats/prometheus
         prometheus.io/port: "15020"
         prometheus.io/scrape: "true"
-        sidecar.istio.io/interceptionMode: REDIRECT
```

Comment:
@stewartbutler @jwendell the injection template has this "PodRedirectAnnot". The intent, I think, was to add these annotations and CNI would read them. As far as I can tell, this logic was only every implemented in kube-inject. Given the vast majority of people do *not* use kube-inject, its not clear to me if the fix is to make the webhook patch add these or if the fix is to stop adding these in kube-inject (as done in this PR).

Please advise

Reply:
Also @knrc I think this impacts the multus CNI stuff

Reply:
Opened https://github.com/istio/istio/issues/26611

Reply:
The real proper fix would be to have Istio CNI follow the same logic as the injector, hopefully by reusing the injection code you're factorizing in this PR. But in the short-term it will need those annotations.